src/main/java/dev/meyi/bn/modules/Module.java

```diff
@@ -1,55 +1,69 @@
 package dev.meyi.bn.modules;
 
-import dev.meyi.bn.config.ModuleConfig;
+import cc.polyfrost.oneconfig.config.annotations.Switch;
+import cc.polyfrost.oneconfig.gui.OneConfigGui;
+import cc.polyfrost.oneconfig.hud.Hud;
+import cc.polyfrost.oneconfig.platform.Platform;
+import dev.meyi.bn.BazaarNotifier;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.Gui;
 import net.minecraft.client.gui.ScaledResolution;
 import org.apache.commons.lang3.StringUtils;
 import org.lwjgl.input.Mouse;
 import org.lwjgl.opengl.Display;
 
-public abstract class Module {
 
-
-  protected int x;
-  protected int y;
-  protected float scale;
-  protected int boundsX, boundsY;
+public abstract class Module extends Hud{
   protected int shift = 0;
   protected boolean moving = false;
-  protected boolean active = true;
   protected boolean needsToMove = false;
   protected int mouseWheelShift = 0;
   protected int padding = 3;
   private int lastMouseX, lastMouseY;
 
+  protected transient String longestString = "";
+
+
+
+  @Switch(name="Show Modules Outside of Bazaar",
+  category = "General"
+  )
+  protected boolean showEverywhere = false;
+
   public Module() {
-    this.x = 0;
-    this.y = 0;
-    this.scale = 1;
+    position.setPosition(20,20);
+    setScale(1, false);
   }
 
-  public Module(ModuleConfig module) {
-    this.x = module.x;
-    this.y = module.y;
-    this.scale = module.scale;
-    this.active = module.active;
+  @Override
+  protected boolean shouldShow() {
+    if (showInGuis && Platform.getGuiPlatform().getCurrentScreen() != null &&
+            (Platform.getGuiPlatform().getCurrentScreen() instanceof OneConfigGui))  return false;
+    if(enabled) {
+      if (BazaarNotifier.inBazaar) return true;
+      if (showInChat && Platform.getGuiPlatform().isInChat()) return true;
+      if (showInDebug && Platform.getGuiPlatform().isInDebug()) return true;
+      if (showInGuis && Platform.getGuiPlatform() != null && Minecraft.getMinecraft().currentScreen != null &&
+              !Platform.getGuiPlatform().isInDebug() && !Platform.getGuiPlatform().isInChat()) return true;
+      return showEverywhere;
+    }else {
+      return false;
+    }

```

Comment:
This is what I was talking about. 
Looking at it now it may just be inBazaar? perhaps that could be put in the render method as well just to exit faster if `inBazaar` is true.

idk if that'd mess up anything else. If it does, just ignore what i've said lol

Reply:
There is still the module specific enabled value. So it's not necessarily better to put the method in the Module list even ignoring the fact that it would be a duplicate since shouldDraw needs to be overwritten to get the expected behavior for everything drawn by OneConfig.